roles/lib_dyn/library/dyn_record.py

```diff
@@ -0,0 +1,444 @@
+  user_password:
+    description:
+      - "The Dyn user's password to log in with. If not set the value of the"
+      - "c(DYNECT_PASSWORD) environment variable is used."
+    required: false
+    default: null
+
+  zone:
+    description:
+      - "The DNS zone in which your record is located."
+    required: true
+    default: null
+
+  record_fqdn:
+    description:
+      - "Fully qualified domain name of the record name to get, create,
+      delete,"
+      - "or update."
+    required: true
+    default: null
+
+  record_type:
+    description:
+      - "Record type."
+    required: true
+    choices: [ 'A', 'AAAA', 'CNAME', 'PTR', 'TXT' ]
+    default: null
+
+  record_value:
+    description:
+      - "Record value. If record_value is not specified; no changes will be"
+      - "made and the module will fail"
+    required: false
+    default: null
+
+  record_ttl:
+    description:
+      - 'Record's "Time to live".  Number of seconds the record remains cached'
+      - 'in DNS servers or c(0) to use the default TTL for the zone.'
+      - 'This option is mutually exclusive with use_zone_ttl'
+    required: false
+    default: 0
+
+  use_zone_ttl:
+    description:
+      - 'Use the DYN Zone's Default TTL'
+      - 'This option is mutually exclusive with record_ttl'
+    required: false
+    default: false
+    mutually exclusive with: record_ttl
+
+notes:
+  - The module makes a broad assumption that there will be only one record per
+  "node" (FQDN).
+  - This module makes an assumptions that record types will not be changed
+  (ie. CNAME to AAAA). For this, first delete then create the new record.
+  - This module returns record(s) in the "result" element when 'state' is set
+  to 'present'. This value can be be registered and used in your playbooks.
+
+requirements: [ dyn ]
+author: "Russell Harrison"
+
+"""
+
+EXAMPLES = '''
+# Attempting to cname www.example.com to web1.example.com
+- name: Update CNAME record
+  dyn_record:
+    state: present
+    record_fqdn: www.example.com
+    zone: example.com
+    record_type: CNAME
+    record_value: web1.example.com
+    record_ttl: 7200
+
+# Use the zones default TTL
+- name: Update CNAME record
+  dyn_record:
+    state: present
+    record_fqdn: www.example.com
+    zone: example.com
+    record_type: CNAME
+    record_value: web1.example.com
+    use_zone_ttl: true
+
+- name: Update A record
+  dyn_record:
+    state: present
+    record_fqdn: web1.example.com
+    zone: example.com
+    record_value: 10.0.0.10
+    record_type: A
+'''
+
+try:
+    IMPORT_ERROR = False
+    # Ansible tends to need a wild card import so we'll use it here
+    # pylint: disable=redefined-builtin, unused-wildcard-import, wildcard-import,
+    # locally-disabled
+    from ansible.module_utils.basic import AnsibleModule
+    from dyn.tm.session import DynectSession
+    from dyn.tm.zones import Zone
+    import dyn.tm.errors
+    import os
+
+except ImportError as error:
+    IMPORT_ERROR = str(error)
```

Comment:
Needs `# pylint: disable=redefined-variable-type` comment before that, as pylint doesn't like redefining var type

Reply:
Yes.  It's an Ops blocker.  See [bugzilla 1569631](https://bugzilla.redhat.com/show_bug.cgi?id=1569631)